### ISSUE TITLE
Feature Request: client_secret in or from apiintegrations

### ISSUE THREAD

**Issue by danjamf** [deploymenttheory/terraform-provider-jamfpro#268]
I see a couple of ways of doing this:-

1) split out as another resource/ds. API Integration is required for the generation of client_secret.

2) add within existing API Integration resource/ds. Optional but will add additional calls if flag set to true and might go against the atomic nature of what other resources do. [PoC](https://github.com/deploymenttheory/terraform-provider-jamfpro/compare/main...danjamf:terraform-provider-jamfpro:client_secret) but would need additional logic on read/update/object etc

**Comment by danjamf** [comment]
Although there is [no Read/Update/Delete](https://developer.jamf.com/jamf-pro/reference/postcreateclientcredentials) for client_secret, which would mean making it its own resource very weak.

**Comment by ShocOne** [comment]
hi @danjamf , what's the usecase for this scenario ? thanks

**Comment by danjamf** [comment]
Taking the client_secret and passing into other providers or tools for integration. Eg Jamf Security Cloud.

**Comment by ShocOne** [comment]
I'll take a look at this next week. You can .source fields in hcl, i just need to run some tests on what the api returns for this field.

**Comment by danjamf** [comment]
Thanks. I'm fairly happy with the functionality of the PoC I posted. Able to use the variable referenced across with other providers etc. I'm just unsure on the design decision of having this additional create inside a resource and whether to allow it with an update too etc.

**Comment by ShocOne** [comment]
Can i ask why during your workflow you would not use a credential manager to store your client id and secret, and then reference them during your pipeline as .hcl env vars ? Since you are doing this anyway to run this provider, why not have a second one for sharing between this and other providers as needed ? This would be more secure and avoids storing the client secret in state which in some cases means clear text.

**Comment by danjamf** [comment]
Fair point! I think if a client_secret can be generated, we leave it to the best practices of the pipeline owner of where and how they store secrets in general.

**Comment by danjamf** [comment]
I looked into how the AWS provider does this with IAM secrets. They do have the option of an [unencrypted output](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_access_key#secret) but also note the preferred option of providing a `pgp_key` and obtaining an [encrypted secret](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_access_key#encrypted_secret). 
A working example using their [vault helper service ](https://github.com/hashicorp/terraform-provider-aws/blob/695458acc50c3c38c0dd3e2dc0c2d363219c0dae/internal/service/iam/encryption.go)is provided for IAM and their Lightsail service (exporting the generated SSH key).
Might this be an appropriate approach?

**Comment by danjamf** [comment]
added in https://github.com/deploymenttheory/terraform-provider-jamfpro/commit/ea09d013ee96a3fba79ac86066b80966fa524675
